tasks/sync_pull.py

```diff
@@ -432,11 +432,12 @@ def trigger_process_flakes(
         self, repoid: int, pull_head: str, branch: str, current_yaml: UserYaml
     ):
         # but only if flake processing is enabled for this repo
-        if FLAKY_TEST_DETECTION.check_value(
-            identifier=repoid, default=False
-        ) and read_yaml_field(
-            current_yaml, ("test_analytics", "flake_detection"), False
-        ):
+        if (
+            FLAKY_TEST_DETECTION.check_value(identifier=repoid, default=False)
+            and read_yaml_field(
+                current_yaml, ("test_analytics", "flake_detection"), False
+            )
+        ) or (FLAKY_SHADOW_MODE.check_value(identifier=repoid, default=False)):
```

Comment:
This `if` statement is kind of hard to read. I would suggest pulling up the conditional into a variable or function. If you pull it up into a function, we can even make this reusable in `test_results_finisher.py`.

Reply:
addressed